src/main/java/org/sagebionetworks/bridge/exporter/handler/SynapseExportHandler.java

```diff
@@ -155,53 +160,129 @@ private synchronized TsvInfo initTsvForTask(ExportTask task) throws BridgeExport
-            synapseTableId = createNewTable(task);
+            serverColumnList = createTableAndGetColumnList(task, columnDefList);
+        } else {
+            serverColumnList = updateTableAndGetColumnList(synapseTableId, columnDefList);
         }
-        return getColumnNameListForTable(synapseTableId);
+
+        // Extract column names from column models
+        List<String> columnNameList = new ArrayList<>();
+        //noinspection Convert2streamapi
+        for (ColumnModel oneServerColumn : serverColumnList) {
+            columnNameList.add(oneServerColumn.getName());
+        }
+        return columnNameList;
     }
 
-    // Helper method to create the new Synapse table. Returns the Synapse table ID.
-    private String createNewTable(ExportTask task) throws BridgeExporterException,
-            SynapseException {
+    // Helper method to create the new Synapse table. Returns the list of columns, with column IDs.
+    private List<ColumnModel> createTableAndGetColumnList(ExportTask task, List<ColumnModel> columnDefList)
+            throws BridgeExporterException, SynapseException {
         ExportWorkerManager manager = getManager();
-
-        // Construct column definition list. Merge COMMON_COLUMN_LIST with getSynapseTableColumnList.
-        List<ColumnModel> columnList = new ArrayList<>();
-        columnList.addAll(COMMON_COLUMN_LIST);
-        columnList.addAll(getSynapseTableColumnList());
+        SynapseHelper synapseHelper = manager.getSynapseHelper();
 
         // Delegate table creation to SynapseHelper.
         long dataAccessTeamId = manager.getDataAccessTeamIdForStudy(getStudyId());
         long principalId = manager.getSynapsePrincipalId();
         String projectId = manager.getSynapseProjectIdForStudyAndTask(getStudyId(), task);
         String tableName = getDdbTableKeyValue();
-        String synapseTableId = manager.getSynapseHelper().createTableWithColumnsAndAcls(columnList, dataAccessTeamId,
+        String synapseTableId = synapseHelper.createTableWithColumnsAndAcls(columnDefList, dataAccessTeamId,
                 principalId, projectId, tableName);
 
         // write back to DDB table
         manager.setSynapseTableIdToDdb(task, getDdbTableName(), getDdbTableKeyName(), getDdbTableKeyValue(),
                 synapseTableId);
 
-        return synapseTableId;
+        // Get the column list.
+        return synapseHelper.getColumnModelsForTableWithRetry(synapseTableId);
     }
 
-    // Helper method to get the column name list for a Synapse table. This queries the column model list from Synapse.
-    private List<String> getColumnNameListForTable(String synapseTableId) throws SynapseException {
-        // Get columns from Synapse
+    // Helper method to detect when a schema changes and updates the Synapse table accordingly. Will reject schema
+    // changes that delete or modify columns. Optimized so if no columns were inserted, it won't modify the table.
```

Comment:
What happens if it rejects a delete or a type change? The export fails, or is the export later put in a new table?